lib/braintree/gateway.js

```diff
@@ -90,12 +91,17 @@ class Gateway {
     };
   }
 
-  pagingFunctionGenerator(search, url, SubjectType, pagedResultsKey, getSubject) {
+  pagingFunctionGenerator(search, url, SubjectType, pagedResultsKey, getSubject, options) {
     return (ids, callback) => {
-      search.ids().in(ids);
       let gateway = this.gateway;
+      let advancedSearch = search;
 
-      gateway.http.post(`${this.config.baseMerchantPath()}/` + url + '/advanced_search', {search: search.toHash()}, function (err, response) {
+      if (options && options.idsOnlyPagination === true) {
+        advancedSearch = new TransactionSearch();
```

Comment:
Is there a reason this can't be done in the transaction_gateway's paging function generator instead? Why bring in logic specific to transaction searches for the generic version?

Reply:
👍
Moved it to `transaction_gateway` now.